### DISCUSSION TITLE
Impersonation using Sanctum

### DISCUSSION THREAD

**Discussion by michaelhume** [laravel/nova-issues#3893] [General]
Hi All,

Considered submitting this as an issue - but just wanted to make sure I'm not missing something. When I attempt to use the impersonation feature, I am redirected to the login page if I attempt to access guarded routes. Discord mentioned that this feature only works with session based auth (I'm using sanctum). 

Is there a way to get this to work using non-session based auth?

**Comment by crynobone** [comment]
Hi, yes Impersonates Users feature requires session-based auth and at the moment we are not looking to expand this feature. However, you can rebind `Laravel\Nova\Contracts\ImpersonatesUsers` with custom implementations that could allow non-session-based auth.
